qiskit_machine_learning/algorithms/classifiers/neural_network_classifier.py

```diff
@@ -119,4 +125,26 @@ def predict(self, X: np.ndarray) -> np.ndarray:  # pylint: disable=invalid-name
     def score(
         self, X: np.ndarray, y: np.ndarray, sample_weight: Optional[np.ndarray] = None
     ) -> float:
+        y = self._encode_categorical(y)
         return ClassifierMixin.score(self, X, y, sample_weight)
+
+    def _fit_and_encode_categorical(self, y: np.ndarray) -> np.ndarray:
+        """Fits label or one-hot encoder and converts categorical target data."""
+
+        if isinstance(y[0], str):
+            # string data is assumed to be categorical
+            y = y.reshape(-1, 1)
```

Comment:
Are we sure that the second dimension in the reshape is `1`?

Reply:
I suppose the implicit assumption in these functions is that if the provided labels are multidimensional then any encoding has already been done. This is a result of checking if `y[0]` is a `string` which, if satisfied, suggests a 1D list of data is provided and can therefore be transformed to a column. I did actaully find a small bug thanks to this comment and will push a change to address it (reshaping to a column throws a warning when using the `LabelEncoder`).